operator/controllers/keptntaskdefinition/reconcile_function.go

```diff
@@ -0,0 +1,109 @@
+package keptntaskdefinition
+
+import (
+	"context"
+	"fmt"
+	klcv1alpha1 "github.com/keptn-sandbox/lifecycle-controller/operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"reflect"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (r *KeptnTaskDefinitionReconciler) ReconcileFunction(ctx context.Context, req ctrl.Request, definition *klcv1alpha1.KeptnTaskDefinition) error {
+	if definition.Spec.Function.InlineReference != (klcv1alpha1.InlineReference{}) {
```

Comment:
would it make sense to make bith the InlineReference and ConfigMapReference a pointer, so we can check for `!= nil` instead?